invenio_communities/communities/services/results.py

```diff
@@ -57,46 +90,39 @@ def total(self):
     @property
     def hits(self):
         """Iterator over the hits."""
-        for hit in self._results.items:
+        for record in self._results.items:
+
             # Project the record
             projection = self._schema.dump(
-                hit,
+                record,
                 context=dict(
                     identity=self._identity,
+                    record=record,
                 ),
             )
             if self._links_item_tpl:
-                projection["links"] = self._links_item_tpl.expand(hit)
+                projection["links"] = self._links_item_tpl.expand(
+                    record, identity=self._identity
+                )
 
             yield projection
 
-    @property
-    def pagination(self):
-        """Create a pagination object."""
-        return Pagination(
-            self._params["size"],
-            self._params["page"],
-            self.total,
-        )
-
-    def to_dict(self):
-        """Return result as a dictionary."""
-        # TODO: This part should imitate the result item above. I.e. add a
-        # "data" property which uses a ServiceSchema to dump the entire object.
 
-        res = {
-            "hits": {
-                "hits": list(self.hits),
-                "total": self.total,
-            }
-        }
+class CommunityItem(RecordItem):
+    """Single request result."""
 
-        if self._params:
-            if self._links_tpl:
-                res["links"] = self._links_tpl.expand(self.pagination)
+    @property
+    def links(self):
+        """Get links for this result item."""
+        return self._links_tpl.expand(self._record, identity=self._identity)
 
+    def to_dict(self):
+        """Get a dictionary for the request."""
+        res = self.data
+        if self._errors:
+            res["errors"] = self._errors
         return res
 
 
-class CommunityFeaturedList(BaseServiceListResult):
-    """List of featured community entry result items."""
+class FeaturedCommunityItem(CommunityItem):
```

Comment:
```suggestion
class CommunityFeaturedItem(CommunityItem):
```

minor: for consistency, though I highly dislike this name ... on the other hand, changing `CommunityFeaturedList` to `FeaturedCommunityList` is backward incompatible (but doable via an alias + deprecation warning).